pkg/config/environment_detection.go

```diff
@@ -71,22 +71,24 @@ func detectContainerFeatures() {
 		detectedFeatures[Docker] = struct{}{}
 	} else {
 		for _, defaultDockerSocketPath := range getDefaultDockerPaths() {
-			if _, err := os.Stat(defaultDockerSocketPath); err == nil {
+			if checkSocketExists(defaultDockerSocketPath) {
 				detectedFeatures[Docker] = struct{}{}
+
 				// Even though it does not modify configuration, using the OverrideFunc mechanism for uniformity
 				AddOverrideFunc(func(Config) {
-					os.Setenv("DOCKER_HOST", "unix://"+defaultDockerSocketPath)
+					os.Setenv("DOCKER_HOST", getDefaultDockerSocketType()+defaultDockerSocketPath)
```

Comment:
nit: Ideally we would use [URL](https://golang.org/pkg/net/url/#URL) object for composing these but it's effectively doing the same thing as the `+` so not a big deal.

Reply:
It would require URLParse, err check + path.Join and there's indeed no gain as it would not even solve missing / double `/`